configure.py

```diff
@@ -1420,6 +1424,106 @@ def bakefile_test_sources(file, sources):
 
     f.close()
 
+def gen_cmake(build_config, using_mods, cc, options):
+    def create_basic_rules(sources):
+        module = {'sources': {}, 'frameworks': set(), 'libs': set()}
+        for source in sources:
+            module['sources'][source] = {'isa_flags': set()}
+        return module
+
+    def update_rules(rules, using_mod):
+        needed = False
+        for source_path in rules['sources']:
+            for mod_source in using_mod.source:
+                if source_path == mod_source :
+                    needed = True
+                    for isa in using_mod.need_isa:
+                        rules['sources'][source_path]['isa_flags'].add(cc.isa_flags[isa])
+        if needed == True:
+            if options.os in using_mod.libs:
+                for lib in using_mod.libs[options.os]:
+                    rules['libs'].add(lib)
+            if options.os in using_mod.frameworks:
+                for framework in using_mod.frameworks[options.os]:
+                    rules['frameworks'].add(framework)
+
+    def generate_sources_list(fd, name, rules):
+        fd.write('set(' + name + '\n')
+        for source in rules['sources']:
+            fd.write('    ${CMAKE_CURRENT_LIST_DIR}/' + source[2:] + '\n')
+        fd.write(')\n\n')
+
+    def generate_special_properties(fd, rules):
+        for source in rules['sources']:
+            joined = ' '.join(rules['sources'][source]['isa_flags'])
+            if joined:
+                fd.write('set_source_files_properties(${CMAKE_CURRENT_LIST_DIR}/' + source[2:] + ' PROPERTIES COMPILE_FLAGS '+ joined + ')\n')
+
+    lib_module = create_basic_rules(build_config.sources)
+    test_module = create_basic_rules(build_config.test_sources)
+    cli_module = create_basic_rules(build_config.cli_sources)
+
+    for module in using_mods:
+        update_rules(lib_module, module)
+        update_rules(test_module, module)
+        update_rules(cli_module, module)
+
+    f = open('CMakeLists.txt', 'w')
+    f.write('cmake_minimum_required(VERSION 3.0.0)\n')
```

Comment:
Yes, It should be not problem - I need only check if in 2.8.* are avaliable: 
* set_source_files_properties, 
* target_link_libraries,
* target_compile_options,
* target_include_directories,
* set_target_properties

As these are very useful to control target dependencies(by "smart" inheritance).